registry-plugin/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// pluginName is the plugin name
+var pluginName = "registry"
+
+// HandlerRegisterer is the symbol the plugin loader will try to load. It must implement the Registerer interface
+var HandlerRegisterer = registerer(pluginName)
+
+type registerer string
+
+func (r registerer) RegisterHandlers(f func(
+	name string,
+	handler func(context.Context, map[string]interface{}, http.Handler) (http.Handler, error),
+)) {
+	f(string(r), r.registerHandlers)
+}
+
+func (r registerer) registerHandlers(_ context.Context, extra map[string]interface{}, h http.Handler) (http.Handler, error) {
+
+	config, ok := extra[pluginName].(map[string]interface{})
+	if !ok {
+		return h, errors.New("configuration not found")
+	}
+
+	hostport, _ := config["hostport"].(string)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		// If the URL path contains /v2/, indicating a request to Distribution Registry HTTP API V2,
+		// the traffic is hijacked and directed to the registry
+		if strings.Contains(req.URL.Path, "/v2/") {
+			req.URL.Scheme = "http"
+			req.URL.Host = hostport
+			req.RequestURI = ""
+
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			// Copy headers, status codes, and body from the backend to the response writer
+			for k, hs := range resp.Header {
+				for _, h := range hs {
+					w.Header().Add(k, h)
+				}
+			}
+			w.WriteHeader(resp.StatusCode)
+			io.Copy(w, resp.Body)
+			resp.Body.Close()
+		} else {
+			h.ServeHTTP(w, req)
+		}
```

Comment:
Less indentation -> more clarity

```suggestion
		if !strings.Contains(req.URL.Path, "/v2/") {
			h.ServeHTTP(w, req)
		}

		// If the URL path contains /v2/, indicating a request to Distribution Registry HTTP API V2,
		// the traffic is hijacked and directed to the registry
		req.URL.Scheme = "http"
		req.URL.Host = hostport
		req.RequestURI = ""

		resp, err := http.DefaultClient.Do(req)
		if err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}

		// Copy headers, status codes, and body from the backend to the response writer
		for k, hs := range resp.Header {
			for _, h := range hs {
				w.Header().Add(k, h)
			}
		}
		w.WriteHeader(resp.StatusCode)
		io.Copy(w, resp.Body)
		resp.Body.Close()
```